### PR TITLE
Convert views to using esbuild.js w/ plugins

### PR DESCRIPTION
**Pull request by leeweisberger**
## Description
Convert views to using the same esbuild script that tasks use which includes default plugins rather than the esbuild CLI.

A user was running into an issue where their view wouldn't deploy when importing a 3p css library. I fixed this earlier in our local discovery/tasks code, but did not fix it in our views code. Switch views to using esbuild which already contains a plugin that removes css files during discovery.

## Test plan
I wrote a unit test that reproduced the issue, and now it works.

---
Note that I'm going to be making a few more improvements to views build. Not trying to make them all at once

### CODE REVIEW

File: pkg/build/package.json [hunk 1]
```diff
@@ -1,8 +1,13 @@
 {
     "dependencies": {
+        "@airplane/views": "1.14.3",
```
Comment (leeweisberger) [inline pkg/build/package.json:3]: Store views dependencies here so that they can be handled by renovate. We're pinning the dependencies that we hope to control and ^ dependencies that we care less about having exact versions of

File: pkg/build/view.go [hunk 2]
```diff
@@ -218,33 +218,19 @@ func viewBundle(root string, buildContext BuildContext, options KindOptions, fil
...
-		}
-		if devDepsOk {
-			if _, ok := devDeps[pkg]; ok {
-				hasPackage = true
-			}
-		}
-		if !hasPackage {
-			packagesToAdd = append(packagesToAdd, pkg)
-		}
+	packagesToAdd, err := getRequiredViewsDependencies([]string{packageJSONPath})
```
Comment (leeweisberger) [inline pkg/build/view.go:221]: This is just a refactor which pulls the versions from the package.json and puts the code in a helper function

File: pkg/build/view.go [hunk 3]
```diff
@@ -399,6 +393,36 @@ func viewBundle(root string, buildContext BuildContext, options KindOptions, fil
...
+
+	requiredDepsMap := map[string]string{
+		"@airplane/views":      buildToolsPackageJSON.Dependencies["@airplane/views"],
+		"react":                buildToolsPackageJSON.Dependencies["react"],
+		"react-dom":            buildToolsPackageJSON.Dependencies["react-dom"],
+		"vite":                 buildToolsPackageJSON.Dependencies["vite"],
+		"esbuild":              buildToolsPackageJSON.Dependencies["esbuild"],
+		"typescript":           buildToolsPackageJSON.Dependencies["typescript"],
+		"@vitejs/plugin-react": buildToolsPackageJSON.Dependencies["@vitejs/plugin-react"],
+	}
```
Comment (ameyakhare) [inline pkg/build/view.go:416]: maybe refactor this into a list and range?

File: renovate.json [hunk 4]
```diff
@@ -12,12 +12,8 @@
   ],
   "packageRules": [
     {
-      "matchPackageNames": ["jsdom"],
-      "schedule": ["before 5am on the first Thursday of the month"]
-    },
-    {
```
Comment (ameyakhare) [inline renovate.json:18]: why was this removed?

**Review by ameyakhare — APPROVED** [review]

File: renovate.json (hunk 4 above)
Comment (leeweisberger) [reply renovate.json:18]: It was invalid. Unrelated. Just looping it in
